Lib/Inc/CParser.h

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file CParser.h
+ *
+ */
+/*
+ *  Created on: Apr 4, 2022
+ *      Author: niuslar
+ */
+
+#ifndef CPARSER_H_
+#define CPARSER_H_
+
+#include "../etl/string.h"
+#include "../etl/vector.h"
+#include "CFIFOBuffer.h"
+#include "IComChannel.h"
+#include "ICommand.h"
+#include "main.h"
+
+#define MAX_TOKENS 60
+
+#ifndef MAX_STRING_SIZE
+#    define MAX_STRING_SIZE 100
+#endif
+
+class CParser : public ICommand
+{
+public:
+    typedef enum TOKEN_TYPES
+    {
+        WHITESPACE,
+        INTEGER,
+        FLOAT,
+        STRING_LITERAL,
+        BEGIN_OPERATOR,
+        END_OPERATOR,
+        START_ARRAY_OP,
+        END_ARRAY_OP,
+        ARGUMENT_DELIMITER,
+        KEY_VALUE_DIV
+    } token_type_t;
+
+    typedef enum PARSING_STATE
+    {
+        IDLE,
+        BEGIN_KEY,
+        END_KEY,
+        BEGIN_VALUE,
+        BEGIN_ARRAY,
+        ADD_VALUE,
+        VALUE_ADDED,
+        END_VALUES,
+        COMMAND_OK,
+        ERROR_MAX_ARGS,
+        ERROR_NESTED_JSON,
+        ERROR
+    } parser_state_t;
+
+    typedef struct token
+    {
+        token_type_t type = WHITESPACE;
+        etl::string<MAX_STRING_SIZE> text;
+
+    } token_t;
+
+    // Public methods
+    CParser();
```

Comment:
ctor should be at the top of the file straight after class definition. This is the thing that is looked up the most often by the user of the class and scrolling down takes time.